lib/archive.ml

```diff
@@ -6,6 +6,20 @@
 
 open Bos_setup
 
+(* Safe wrapping for some Bytes functions *)
+
+module Sbytes = struct
+  include Bytes
+
+  let blit_string src srcoff dst dstoff len ~error =
+    try R.ok (blit_string src srcoff dst dstoff len)
+    with Invalid_argument _ -> R.error_msg error
```

Comment:
How about using `Invalid_argument`'s argument as a message here and use a `Result.map_error` function when we want to override it?

The `~error` argument feels a bit odd especially if we move this module elsewhere so it can be reused.

I agree it encourages us to think about what the error means when it arises but in some places it might not make sense and just get in the way to have to provide that msg.

Reply:
I agree, done